legend-engine-pure/legend-engine-pure-code/legend-engine-pure-code-compiled-core/src/main/resources/core/pure/corefunctions/collectionExtension.pure

```diff
@@ -123,6 +123,11 @@ function meta::pure::functions::collection::and(vals:Boolean[*]):Boolean[1]
    $vals->fold({i,a|$i && $a}, true)
 }
 
+function meta::pure::functions::collection::or(vals:Boolean[*]):Boolean[1]
+{
+   $vals->fold({i,a|$i || $a}, false)
```

Comment:
seems duplicated from the code below... can one delegate to the other?

Reply:
a simple delegation will not work due to recursion, the alternative would be to introduce a different statement such as
if ($vals->isEmpty(), | false, | $values->toOneMany()->or()) which feels less optimal than this simple duplication.
The idea would be to remove the 1..* function but requires some further analysis first.